src/common/src/catalog/mod.rs

```diff
@@ -210,6 +212,59 @@ impl From<&TableId> for risingwave_pb::plan_common::TableRefId {
     }
 }
 
+// TODO: TableOption is deplicated with the properties in table catalog, We can refactor later to
+// directly fetch such options from catalog when creating compaction jobs.
+#[derive(Clone, Debug, PartialEq, Default, Copy)]
+pub struct TableOption {
+    pub ttl: Option<u32>,
+}
+
+impl From<&risingwave_pb::hummock::TableOption> for TableOption {
+    fn from(table_option: &risingwave_pb::hummock::TableOption) -> Self {
+        let ttl = if table_option.ttl == hummock::TABLE_OPTION_DUMMY_TTL {
+            None
+        } else {
+            Some(table_option.ttl)
+        };
+
+        Self { ttl }
+    }
+}
+
+impl From<&TableOption> for risingwave_pb::hummock::TableOption {
+    fn from(table_option: &TableOption) -> Self {
+        Self {
+            ttl: table_option.ttl.unwrap_or(0),
```

Comment:
-> `unwrap_or(TABLE_OPTION_DUMMY_TTL)`

We use 0 as the special value. Do we need to disallow user to set ttl to 0?

Reply:
its a good idea to disallow user to set ttl to 0 in frontend (but now frontend had not distinguish with_option value type)  even if the logic is masked in the storage layer. 
conculution: if the ttl be set to 0 , we can cover it in the storage layer